src/components/encoding-pane/property-editor-schema.ts

```diff
@@ -63,81 +92,217 @@ const POSITION_FIELD_QUANTITATIVE_INDEX = {
       };
       return generateSchema(prop, nestedProp, DEFAULT_SELECT_UISCHEMA, propTab, schemaProperty);
+    } else if (nestedProp === 'range' || nestedProp === 'domain') {
+      const schemaProperty: StringSchema = {
+        ...baseUiSchema
+      };
+      return generateSchema(prop, nestedProp, isDiscrete(fieldDef) ?
+        DEFAULT_DISCRETE_SCALE_ARRAY_UISCHEMA : DEFAULT_CONTINUOUS_SCALE_ARRAY_UISCHEMA, propTab, schemaProperty);
     }
   } else if (prop === 'axis') {
     if (nestedProp === 'orient') {
       const schemaProperty: StringSchema = {
-        title: title,
+        ...baseUiSchema,
         enum: (vlSchema as any).definitions.AxisOrient.enum,
-        type: 'string'
       };
       return generateSchema(prop, nestedProp, DEFAULT_SELECT_UISCHEMA, propTab, schemaProperty);
     } else if (nestedProp === 'title') {
       const schemaProperty: StringSchema = {
-        title: title,
-        type: 'string'
+        ...baseUiSchema
       };
       return generateSchema(prop, nestedProp, DEFAULT_TEXT_UISCHEMA, propTab, schemaProperty);
     }
   } else if (prop === 'stack') {
     const schemaProperty: StringSchema = {
-      title: title,
+      ...baseUiSchema,
       enum: (vlSchema as any).definitions.StackOffset.enum,
-      type: 'string'
     };
     return generateSchema(prop, nestedProp, DEFAULT_SELECT_UISCHEMA, propTab, schemaProperty);
+  } else if (prop === 'size') {
+    const schemaProperty: StringSchema = {
+      ...baseUiSchema
+    };
+    return generateSchema(prop, nestedProp, DEFAULT_TEXT_UISCHEMA, propTab, schemaProperty);
+  } else if (prop === 'legend') {
+    if (nestedProp === 'orient') {
+      const schemaProperty: StringSchema = {
+        ...baseUiSchema,
+        enum: (vlSchema as any).definitions.LegendOrient.enum
+      };
+      return generateSchema(prop, nestedProp, DEFAULT_SELECT_UISCHEMA, propTab, schemaProperty);
+    } else if (nestedProp === 'title') {
+      const schemaProperty: StringSchema = {
+        ...baseUiSchema
+      };
+      return generateSchema(prop, nestedProp, DEFAULT_TEXT_UISCHEMA, propTab, schemaProperty);
+    } else if (nestedProp === 'type') {
+      const schemaProperty: StringSchema = {
+        ...baseUiSchema,
+        enum: (vlSchema as any).definitions.Legend.properties.type.enum,
+      };
+      return generateSchema(prop, nestedProp, DEFAULT_SELECT_UISCHEMA, propTab, schemaProperty);
+    }
+  } else if (prop === 'format') {
+    const schemaProperty: StringSchema = {
+      ...baseUiSchema
+    };
+    return generateSchema(prop, nestedProp, DEFAULT_TEXT_UISCHEMA, propTab, schemaProperty);
   } else {
     throw new Error('Property combination not recognized');
   }
 }
 
 
+// TODO: Eventually refactor to Vega-Lite
+function getSupportedScaleTypes(shelfId: ShelfId, fieldDef: ShelfFieldDef): string[] {
+  let scaleTypes: string[] = [];
+  if (contains([Channel.X, Channel.Y], shelfId.channel)) {
```

Comment:
Better check switch case by data type first, then do if else for channel inside.  The method will be way less redundant.